src/function_3.py

```
@@ -0,0 +1,56 @@
+                img = cv2.imread(img_path)
+                if img is None:
+                    continue
+
+                gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+                faces = face_cascade.detectMultiScale(gray, scaleFactor=1.1, minNeighbors=5)
+
+                for i, (x, y, w, h) in enumerate(faces):
+                    face_crop = img[y:y+h, x:x+w]
+
+                    if log_func:
+                            log_func("[INFO] Preprocessing Images...")
```

Flake8 (7.2.0)

[error] over-indented

(E117)